src/main/java/de/tum/in/www1/artemis/config/MetricsBean.java

```diff
@@ -113,13 +165,231 @@ private void registerWebsocketMetrics() {
+    }
+
+    private void registerPublicArtemisMetrics() {
+        SecurityUtils.setAuthorizationObject();
+
+        activeUserMultiGauge = MultiGauge.builder("artemis.statistics.public.active_users").description("Number of active users within the last period, specified in days")
+                .register(meterRegistry);
+
+        Gauge.builder("artemis.statistics.public.active_courses", activeCoursesGauge::get).description("Number of active courses").register(meterRegistry);
+
+        Gauge.builder("artemis.statistics.public.courses", coursesGauge::get).description("Number of courses").register(meterRegistry);
+
+        studentsCourseGauge = MultiGauge.builder("artemis.statistics.public.course_students").description("Number of registered students per course").register(meterRegistry);
+
+        Gauge.builder("artemis.statistics.public.active_exams", activeExamsGauge::get).description("Number of active exams").register(meterRegistry);
+
+        Gauge.builder("artemis.statistics.public.exams", examsGauge::get).description("Number of exams").register(meterRegistry);
+
+        studentsExamGauge = MultiGauge.builder("artemis.statistics.public.exam_students").description("Number of registered students per exam").register(meterRegistry);
+
+        activeExerciseGauge = MultiGauge.builder("artemis.statistics.public.active_exercises").description("Number of active exercises by type").register(meterRegistry);
+
+        exerciseGauge = MultiGauge.builder("artemis.statistics.public.exercises").description("Number of exercises by type").register(meterRegistry);
+    }
+
     private static double extractWebsocketUserCount(WebSocketHandler webSocketHandler) {
         if (webSocketHandler instanceof SubProtocolWebSocketHandler subProtocolWebSocketHandler) {
             return subProtocolWebSocketHandler.getStats().getWebSocketSessions();
         }
         return -1;
     }
 
+    private double getUpcomingDueExercisesCount(int minutes, ExerciseType exerciseType) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return exerciseRepository.countExercisesWithCurrentOrUpcomingDueDateWithinTimeRange(now, endDate, exerciseType.getExerciseClass());
+    }
+
+    private double getUpcomingDueExercisesCountWithStudentMultiplier(int minutes, ExerciseType exerciseType) {
+        SecurityUtils.setAuthorizationObject();
+        var now = ZonedDateTime.now();
+        var endDate = ZonedDateTime.now().plusMinutes(minutes);
+        return exerciseRepository.countStudentsInExercisesWithCurrentOrUpcomingDueDateWithinTimeRange(now, endDate, exerciseType.getExerciseClass());
+    }
+
+    private double getUpcomingDueExercisesCountWithActiveStudentMultiplier(int minutes, ExerciseType exerciseType, int numberOfDaysToCountAsActive) {
```

Comment:
Identical to `getUpcomingReleasedExercisesCountWithActiveStudentMultiplier` below?

Reply:
@b-fein The difference is that one method used the due date and the other one uses the release date (which might make a difference for long-lasting exercises such as homework exercises).